flang/lib/Lower/CharacterRuntime.cpp

```diff
@@ -88,3 +88,20 @@ Fortran::lower::genCharCompare(Fortran::lower::AbstractConverter &converter,
   return genRawCharCompare(converter, loc, cmp, lhsBuffer, fir::getLen(lhs),
                            rhsBuffer, fir::getLen(rhs));
 }
+
+void Fortran::lower::genTrim(Fortran::lower::FirOpBuilder &builder,
+                             mlir::Location loc, const mlir::Value &resultBox,
+                             const mlir::Value &stringBox) {
+  auto trimFunc = getRuntimeFunc<mkRTKey(Trim)>(loc, builder);
+  auto fTy = trimFunc.getType();
+  auto sourceFile = builder.locationToFilename(loc);
+  auto sourceLine = builder.locationToLineNo(loc, fTy.getInput(3));
+
+  llvm::SmallVector<mlir::Value, 4> args;
+  auto i = 0;
+  args.emplace_back(builder.createConvert(loc, fTy.getInput(i++), resultBox));
```

Comment:
Why the post-increment? Seems like 0, 1, 2, 3 would be clearer.

Reply:
Indeed, bad idea. Thanks.